src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -335,34 +335,112 @@ def copy_into_location(
             block=block,
         )
 
-    def csv(self, path: str, overwrite: bool = False, compression: str = None, single: bool = True,
-            partition_by: ColumnOrName = None) -> Union[List[Row], AsyncJob]:
+    def csv(
+        self,
+        path: str,
+        overwrite: bool = False,
+        compression: str = "None",
+        single: bool = True,
```

Comment:
based on https://docs.snowflake.com/en/sql-reference/sql/copy-into-location#type-csv, compression should be "AUTO". Why are other options not added that are available in this link?

Reply:
The same is for json and parquet

Reply:
According to the conversation we had, I reorganized the arguments to be aligned with the copy_into_location arguments.